stubs/SQLAlchemy/sqlalchemy/orm/strategy_options.pyi

```diff
@@ -23,44 +24,116 @@ class Load(Generative, LoaderOption):
     # added dynamically at runtime
-    def contains_eager(self, attr, alias: Incomplete | None = ...): ...
-    def load_only(self, *attrs): ...
-    def joinedload(self, attr, innerjoin: Incomplete | None = ...): ...
-    def subqueryload(self, attr): ...
-    def selectinload(self, attr): ...
-    def lazyload(self, attr): ...
-    def immediateload(self, attr): ...
-    def noload(self, attr): ...
-    def raiseload(self, attr, sql_only: bool = ...): ...
-    def defaultload(self, attr): ...
-    def defer(self, key, raiseload: bool = ...): ...
-    def undefer(self, key): ...
-    def undefer_group(self, name): ...
-    def with_expression(self, key, expression): ...
-    def selectin_polymorphic(self, classes): ...
+    def contains_eager(loadopt, attr, alias: Incomplete | None = ...): ...
+    def load_only(loadopt, *attrs): ...
+    def joinedload(loadopt, attr, innerjoin: Incomplete | None = ...): ...
+    def subqueryload(loadopt, attr): ...
+    def selectinload(loadopt, attr): ...
+    def lazyload(loadopt, attr): ...
+    def immediateload(loadopt, attr): ...
+    def noload(loadopt, attr): ...
+    def raiseload(loadopt, attr, sql_only: bool = ...): ...
+    def defaultload(loadopt, attr): ...
+    def defer(loadopt, key, raiseload: bool = ...): ...
+    def undefer(loadopt, key): ...
+    def undefer_group(loadopt, name): ...
+    def with_expression(loadopt, key, expression): ...
+    def selectin_polymorphic(loadopt, classes): ...
 
 class _UnboundLoad(Load):
     path: Any
     local_opts: Any
     def __init__(self) -> None: ...
 
-class loader_option:
-    name: Any
-    fn: Any
-    def __call__(self, fn): ...
-
-def contains_eager(loadopt, attr, alias: Incomplete | None = ...): ...
-def load_only(loadopt, *attrs): ...
-def joinedload(loadopt, attr, innerjoin: Incomplete | None = ...): ...
-def subqueryload(loadopt, attr): ...
-def selectinload(loadopt, attr): ...
-def lazyload(loadopt, attr): ...
-def immediateload(loadopt, attr): ...
-def noload(loadopt, attr): ...
-def raiseload(loadopt, attr, sql_only: bool = ...): ...
-def defaultload(loadopt, attr): ...
-def defer(loadopt, key, raiseload: bool = ...): ...
-def undefer(loadopt, key): ...
-def undefer_group(loadopt, name): ...
-def with_expression(loadopt, key, expression): ...
-def selectin_polymorphic(loadopt, classes): ...
+###
+# The methods below are decorated with the class loader_option
+# They dynamically become instances of loader_option,
+# wich is callable with their original parameters.
+#
+# While both mypy and pyright's validation work, Pylance is unable to
+# show the parameters and return types.
+#
+# There is a workaround (define the method for Pylance, then reassign
+# an instance of loader_option to it for mypy, and add pyright+Flake8
+# suppressions), but it is too hacky and relies on some unsupported quirks.
+#
+# Asking Pylance to add support for these generic callables might be preferable.
```

Comment:
Don't know about pylance, but pyright is (reasonably) pretty resistant to adding any special-casing for Python libraries outside of the stdlib, so I don't know how likely this is to happen, sadly.

Reply:
pyright is accurate in its errors for parameters type, name, count etc. here. So it looks like something that might've to be handled on Pylance's side to present that information directly to the user.
I'll open a feature request on pylance's side to support this use-case once I can boil it down to a minimal repro. Whether they accept or not this is now accurate and better than it was before (the __call__ signature was misleading anyway given its dynamic nature).

I'll update the inline comment to not mention said workaround, that's more something that belongs as a PR comment like this one.
For posterity, here's the original comment description with some images:

```python
# While both mypy and pyright's validation work, Pylance is unable to
# show the parameters and return types.
#
# There is a workaround (define the method for Pylance, then reassign
# an instance of loader_option to it for mypy, and add pyright+Flake8
# suppressions), but it is too hacky and relies on some unsupported quirks.
#
# Asking Pylance to add support for these generic callables might be preferable.
```

The workaround: https://github.com/python/typeshed/pull/9596#discussion_r1088543062

Pyright spots errors and Pylance picks them up:
![image](https://user-images.githubusercontent.com/1350584/215300591-58a28f81-f8b4-4a07-93b6-f31a6ab55adc.png)

Pylance does not see parameters and docstring:
![image](https://user-images.githubusercontent.com/1350584/215300642-0b09cee8-5422-42a4-a810-87127a99f12c.png)

With workaround:
![image](https://user-images.githubusercontent.com/1350584/215300684-cf010802-e816-4abe-9b03-f056b1b91c80.png)